GirafRest/Controllers/PictogramController.cs

```diff
@@ -276,21 +276,44 @@ public async Task<Response<WeekPictogramDTO>> SetPictogramImage(long id)
             //Update the image
             byte[] image = await _giraf.ReadRequestImage(HttpContext.Request.Body);
 
-            string path = imagePath + pictogram.Id + ".png";
-        
+            string originalPath = imagePath + pictogram.Id + ".png";
+            string tempPath = imagePath + pictogram.Id + "_temp.png";
+            string oldPath = imagePath + pictogram.Id + "_old.png";
+            
+            
             if (image.Length > 0){
-                // Upload new image, change name to id of pictogram
-                System.IO.File.WriteAllBytes(path, new byte[0]);
-                
                 using (FileStream fs =
-                    new FileStream( path,
+                    new FileStream( tempPath,
                         FileMode.OpenOrCreate))
                 {
                     fs.Write(image);
                 }
+            
+                if (System.IO.File.Exists(originalPath) && !System.IO.File.Exists(oldPath))
+                {
+                    System.IO.File.Move(originalPath, oldPath);   
+                    System.IO.File.Move(tempPath, originalPath);
+                }else if (System.IO.File.Exists(originalPath) && System.IO.File.Exists(oldPath))

```

Comment:
```suggestion
                }
                else if (System.IO.File.Exists(originalPath) && System.IO.File.Exists(oldPath))
```

I have a feeling that this curly boy wants to be styled the same way as the others